app/knight/battles.py

```diff
@@ -0,0 +1,24 @@
+def first_battle(lancelot, mordred):
+    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
+    mordred["hp"] -= lancelot["power"] - mordred["protection"]
+
+    if lancelot["hp"] <= 0:
+        lancelot["hp"] = 0
+
+    if mordred["hp"] <= 0:
+        mordred["hp"] = 0
+
+    return lancelot, mordred
+
+
+def second_battle(arthur, red_knight):
```

Comment:
It seems that this function is a copy of the `first_battle` with different names for arguments. It's not ok

Reply:
You can create one function and pass different knights there twice